packages/base-controller/src/BaseController.ts

```diff
@@ -120,17 +120,17 @@ export class BaseController<C extends BaseConfig, S extends BaseState> {
         ? (config as C)
         : Object.assign(this.internalConfig, config);
 
-      for (const key in this.internalConfig) {
+      for (const key of Object.keys(this.internalConfig) as (keyof C)[]) {
         if (typeof this.internalConfig[key] !== 'undefined') {
-          (this as any)[key as string] = this.internalConfig[key];
+          (this as any)[key] = this.internalConfig[key];
         }
       }
```

Comment:
Can we iterate over `Object.entries` instead? (We also don't need to use `typeof` to check for `undefined`, so we can simplify that too.)

``` suggestion
      for (const [key, value] of Object.entries(this.internalConfig)) {
        if (value !== undefined) {
          (this as any)[key] = value;
        }
      }
```

Reply:
Nice! This is much cleaner.